File: LeetCode/0365. Water and Jug Problem/solution.cpp
/**
 * 28 / 28 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.9 MB 
 */
class Solution {
public:
    bool canMeasureWater(int jug1Capacity, int jug2Capacity, int targetCapacity) {
        if (jug1Capacity + jug2Capacity < targetCapacity) return false;
        if (jug1Capacity == 0 || jug2Capacity == 0) return targetCapacity == 0 || jug2Capacity + jug2Capacity == targetCapacity;
        return targetCapacity % __gcd(jug1Capacity, jug2Capacity) == 0;
    }
};

File: LeetCode/0365. Water and Jug Problem/solution.py
"""
28 / 28 test cases passed.
Runtime: 36 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def canMeasureWater(self, x: int, y: int, z: int) -> bool:
        if x + y < z: return False
        if x == 0 or y == 0: return z == 0 or x + y == z
        return z % math.gcd(x, y) == 0
        